src/Aserto.AspNetCore.Middleware/Clients/Directory/V3/IDirectoryAPIClient.cs

```diff
@@ -0,0 +1,195 @@
+        /// </summary>
+        /// <param name="id">The id of the object.</param>
+        /// <param name="type">The type of the object.</param>
+        /// <param name="displayName">The display name of the object.</param>
+        /// <param name="properties">A struct representing the properties bag of the object.</param>
+        /// <param name="hash">The hash of the object.</param>
+        /// <returns>A SetObjectResponse object.</returns>
+        Task<SetObjectResponse> SetObjectAsync(string id, string type, string displayName = "", Struct properties = null, string hash = "");
+
+        /// <summary>
+        /// Deletes an object.
+        /// </summary>
+        /// <param name="id">The id of the object.</param>
+        /// <param name="type">The type of the object.</param>
+        /// <param name="withRelations">Remove the relations for this object also.</param>
+        /// <returns>A DeleteObjectResponse object.</returns>
+        Task<DeleteObjectResponse> DeleteObjectAsync(string id, string type, bool withRelations = false);
+
+        /// <summary>
+        /// Creates a relation if it doesn't exists, updates an existing one.
+        /// </summary>
+        /// <param name="subjectId">The id of the subject.</param>
+        /// <param name="subjectType">The type of the subject.</param>
+        /// <param name="subjectRelation">The relaton of the subject.</param>
+        /// <param name="objId">The id of the object.</param>
+        /// <param name="objType">The type of the object.</param>
+        /// <param name="relationName">The type name of the relation.</param>
+        /// <param name="hash">The hash of the object.</param>
+        /// <returns>A SetRelationResponse object.</returns>
+        Task<SetRelationResponse> SetRelationAsync(string subjectId, string subjectType, string subjectRelation, string objId, string objType, string relationName, string hash = "");
+
+        /// <summary>
+        /// Deletes a relation.
+        /// </summary>
+        /// <param name="subjectId">The Id of the subject.</param>
+        /// <param name="subjectType">The type of the subject.</param>
+        /// <param name="subjectRelation">The relaton of the subject.</param>
+        /// <param name="objId">The Id of the object.</param>
+        /// <param name="objType">The type of the object.</param>
+        /// <param name="relationName">The name of the relation type.</param>
+        /// <returns>A DeleteRelationResponse object.</returns>
+        Task<DeleteRelationResponse> DeleteRelationAsync(string subjectId = "", string subjectType = "", string subjectRelation = "", string objId = "", string objType = "", string relationName = "");
+
+        /// <summary>
+        /// Get the directory manifest.
+        /// </summary>
+        /// <param name="request">Get manifest request.</param>
+        /// <returns>A GetManifestResponse object.</returns>
+        Task<GetManifestResponse> GetManifest(GetManifestRequest request);
```

Comment:
Why do we use the Async suffix in all the reader and writer methods but not in model, importer, and exporter?